src/context/auth/authContextController/AuthContextController.tsx

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react';
+
+import { AuthContext } from '../authContext/AuthContext';
+import { AuthContextControllerProps } from './AuthContextController.types';
+import { checkIsLoggedInAction } from 'api/actions/auth/authActions';
+import { useQuery } from 'api/hooks';
+import { AuthStorage } from '../authStorage.enum';
+
+export const AuthContextController = ({ children }: AuthContextControllerProps) => {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+
+  const { data } = useQuery<string>(checkIsLoggedInAction(), { enabled: !!localStorage.getItem(AuthStorage.TOKEN) });
+
+  useEffect(() => {
+    setIsLoggedIn(!!data);
+  }, [data]);
+
+  return (
+    <AuthContext.Provider
+      value={{
+        isLoggedIn,
+        setToken: (token?: string) => {
```

Comment:
Dobra, to ogólnie tak - nie jest to złe rozwiązanie, ale można to zrobić trochę lepiej.
Typ kontekstu jest jak najbardziej poprawny (isLoggedIn i setToken), niemniej ja bym tu trzymał w stanie nie 0/1 zalogowania, tylko cały token (który jako początkową wartość może jak najbardziej pobrać ją z localStorage), a `isLoggedIn: !!token`. A ustawianie localStorage robił przez useEffect nastawionego na token. To jest trochę bardziej eleganckie, bo to właśnie praca z local storagem jest tutaj side efectem.